### PR TITLE
Filter focusable child elements by it's visibility

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Visibility of focusable elements, missing feature of #6866

### CODE REVIEW

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/persist/persist-focus.service.spec.ts [hunk 1]
```diff
@@ -75,6 +76,45 @@ describe('PersistFocusService', () => {
...
+  describe('clear()', () => {
+    it(`should clear the global persisted value`, () => {
+      const key = 'my-key';
+      service.set(key);
+      expect(service.get()).toEqual('my-key');
+      service.clear();
+      expect(service.get()).toBeUndefined();
+    });
+
+    it(`should clear a value for a specific`, () => {
```
Comment (marlass) [inline projects/storefrontlib/src/layout/a11y/keyboard-focus/persist/persist-focus.service.spec.ts]:
```suggestion
    it(`should clear a value for a specific group`, () => {
```

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/services/select-focus.util.ts [hunk 2]
```diff
@@ -53,14 +52,35 @@ export class SelectFocusUtility {
    * are build from the `focusableSelectors`.
    *
    * @param host the `HTMLElement` used to query focusable elements
-   * @param locked indicates whether inactive (`tabindex="-1"`) focusable elements should be returend as well
+   * @param locked indicates whether inactive (`tabindex="-1"`) focusable elements should be returned
+   * @param invisible indicates whether hidden focusable elements should be returend
```
Comment (marlass) [inline projects/storefrontlib/src/layout/a11y/keyboard-focus/services/select-focus.util.ts]:
```suggestion
   * @param invisible indicates whether hidden focusable elements should be returned
```

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/services/select-focus.util.ts [hunk 3]
```diff
@@ -40,11 +40,10 @@ export class SelectFocusUtility {
   ): HTMLElement {
     const selector =
       typeof config?.autofocus === 'string' ? config.autofocus : '[autofocus]';
-
     // fallback to first focusable
     return (
-      this.query(host, selector).find(Boolean) ||
-      this.findFocusable(host).find(Boolean)
+      this.query(host, selector).find(el => !this.isHidden(el)) ||
+      this.findFocusable(host).find(el => !this.isHidden(el))
```
Comment (marlass) [inline projects/storefrontlib/src/layout/a11y/keyboard-focus/services/select-focus.util.ts]: Isn't it already filtered?

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/services/select-focus.util.spec.ts [hunk 4]
```diff
@@ -66,12 +89,48 @@ describe('SelectFocusUtility', () => {
...
-    it('should not find hidden children', () => {
+    it('should not find attribute-hidden children', () => {
       const host = fixture.debugElement.query(By.css('#d')).nativeElement;
       const button = fixture.debugElement.query(By.css('#d1')).nativeElement;
       expect(service.findFocusable(host).length).toEqual(1);
       expect(service.findFocusable(host)[0]).toEqual(button);
     });
+
+    describe('hide by css', () => {
+      it('should find 4 focusable elements which are not hidden by css', () => {
```
Comment (marlass) [inline projects/storefrontlib/src/layout/a11y/keyboard-focus/services/select-focus.util.spec.ts]:
```suggestion
      it('should find 5 focusable elements which are not hidden by css', () => {
```

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/services/select-focus.util.ts (hunk 3 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/layout/a11y/keyboard-focus/services/select-focus.util.ts]: oh yes you're right 👍

**Review by marlass — APPROVED** [review]

**Pull request merged**